diff --git a/src/core/selinux-setup.c b/src/core/selinux-setup.c
index db14db88ee..c8a9485617 100644
--- a/src/core/selinux-setup.c
+++ b/src/core/selinux-setup.c
@@ -53,6 +53,9 @@ int mac_selinux_setup(bool *loaded_policy) {
         if (selinux_init_load_policy(&enforce) == 0) { /* NB: Apparently doesn't set useful errno! */
                 mac_selinux_retest();
 
+                /* The Nix store does not support extended attributes, so use a static label instead. */
+                const char *static_label = "system_u:system_r:init_t";
+
                 /* Transition to the new context */
                 _cleanup_freecon_ char *label = NULL;
                 r = mac_selinux_get_create_label_from_exe(SYSTEMD_BINARY_PATH, &label);
@@ -60,7 +63,7 @@ int mac_selinux_setup(bool *loaded_policy) {
                         log_open();
                         log_warning_errno(r, "Failed to compute init label, ignoring: %m");
                 } else {
-                        r = RET_NERRNO(setcon_raw(label));
+                        r = RET_NERRNO(setcon_raw(static_label));
                         log_open();
                         if (r < 0)
                                 log_warning_errno(r, "Failed to transition into init label '%s', ignoring: %m", label);
