diff --git a/src/core/selinux-setup.c b/src/core/selinux-setup.c
index 89f736b47b..1da17ea0f5 100644
--- a/src/core/selinux-setup.c
+++ b/src/core/selinux-setup.c
@@ -69,12 +69,15 @@ int mac_selinux_setup(bool *loaded_policy) {
 
                 /* Transition to the new context */
                 r = mac_selinux_get_create_label_from_exe(SYSTEMD_BINARY_PATH, &label);
+
+                /* The Nix store does not support extended attributes, so use a static label instead. */
+                const char *static_label = "system_u:system_r:init_t";
+
                 if (r < 0 || !label) {
                         log_open();
                         log_error("Failed to compute init label, ignoring.");
                 } else {
-                        r = setcon_raw(label);
-
+                        r = setcon_raw(static_label);
                         log_open();
                         if (r < 0)
                                 log_error("Failed to transition into init label '%s', ignoring.", label);
