diff --git a/policy/modules/kernel/kernel.te b/policy/modules/kernel/kernel.te
index 65c814a97..1723083d3 100644
--- a/policy/modules/kernel/kernel.te
+++ b/policy/modules/kernel/kernel.te
@@ -16,6 +16,7 @@ gen_bool(secure_mode_insmod, false)
 attribute can_load_kernmodule;
 attribute can_receive_kernel_messages;
 attribute can_dump_kernel;
+attribute can_exec_unlabeled;
 
 neverallow ~can_load_kernmodule self:capability sys_module;
 
@@ -209,7 +210,7 @@ kernel_rootfs_mountpoint(unlabeled_t)
 fs_associate(unlabeled_t)
 sid file gen_context(system_u:object_r:unlabeled_t,s0)
 sid unlabeled gen_context(system_u:object_r:unlabeled_t,mls_systemhigh)
-neverallow * unlabeled_t:file entrypoint;
+neverallow ~can_exec_unlabeled unlabeled_t:file entrypoint;
 
 # These initial sids are no longer used, and can be removed:
 sid any_socket		gen_context(system_u:object_r:unlabeled_t,mls_systemhigh)
